modules/affile/logproto-file-writer.c

```diff
@@ -135,10 +138,27 @@ log_proto_file_writer_flush(LogProtoClient *s)
   self->buf_count = 0;
   self->sum_len = 0;
 
-  FileFlushSignalData signal_data = {};
+  gchar *file_name = g_strdup_printf("%s", self->filename);
+  gsize file_size = lseek(self->fd, 0, SEEK_END);
```

Comment:
`lseek` might be a cheaper way to check the file's size, but this will be called unconditionally, even if the file-rotation plugin is not configured.

I think it would be better to query the file size in the file-rotation plugin using fstat or stat.
What do you think?

Reply:
But isn't it a more expensive call?

Reply:
Okay it's a 3 vs 1 syscall's difference

Reply:
My only concern is that calling lseek here adds an unnecessary cost for the file destination. We calculate the file size, and most of the time, we do nothing with it, because our users do not always use file rotation.

It also looks a bit odd in the file writer because it is hard to explain why a flush signal needs the file's size in general.

Reply:
Ok then I'll replace it with the fstat implementation.